shells/pipes-shell/source/pipe.js

```diff
@@ -36,8 +36,9 @@ export const busReady = async (bus, {manifest}) => {
 const configureRuntime = async ({rootPath, urlMap, storage, manifest}, bus) => {
   // configure arcs runtime environment
   Runtime.init(rootPath, urlMap);
-  // marshal context
+  // marshal and bind context
   const context = await requireContext(manifest || config.manifest);
+  Runtime.getRuntime().bindContext(context);
```

Comment:
Suggest we add `context` as a third parameter to `Runtime.init`. WDYT?

Reply:
On second thought, this is going to need refactoring in any case, so we can discuss details later.

*Refactoring because I suspect we will want to separate the path-environment concept from the set-of-arcs-in-a-given-context concept.

Reply:
Sound terrific to have better flexibility!

Btw, I tried to add context as an optional third param at the very first place by
invoking requireContext() followed by Runtime.init() turns out the requireContext() raised an exception immediately complaining no loader in place (loader is instantiated at Runtime.init())
The mutual dep. between requireContext and Runtime.init makes me cook the change this way ;-)

Reply:
I came here to suggest the same - add `context` as parameter to `Runtime.init` because Runtime is the object that conceptually has **the** context and it  shouldn't be updated dynamically...

Reply:
Updated the reason why I couldn't do that at above.
I think the refactoring is really required to fix the deps between context generation, loader instantiation and runtime instantiation.

Reply:
sorry, i didn't read carefully enough. 
@sjmiles can we alternatively just pass the manifest into `Runtime.init` and have it do the parsing and set the `context`? if that sounds reasonable, i'm happy to send you a pr.

Reply:
Having a Loader requires having path setup. 
Constructing a Context often requires a Loader.
For these reasons, I think we should tease apart the Path Environment from the Runtime object as I mentioned above. IMO, this should be done deliberately, and I don't see the need for any other tactical changes right away. WDYT?

Reply:
Runtime should never be without a context, and the context should never be reset dynamically later, which means it should be a ctor parameter, not a setter method.
Requiring a loader to create the context is an even stronger argument for passing the manifest string into the Runtime ctor and parsing it there with the newly constructed Loader. WDYT?